src/structures/GuildAuditLogs.js

```diff
@@ -480,8 +483,22 @@ class GuildAuditLogsEntry {
           ),
           guild,
         );
+    } else if (targetType === Targets.CHANNEL) {
+      this.target =
+        guild.channels.cache.get(data.target_id) ||
+        Channel.create(
```

Comment:
This will also insert the channel into the cache of the guild, I don't think this is desirable: 
https://github.com/discordjs/discord.js/blob/32d8c8ba940b31ab577c3de518fef382fe33a60d/src/structures/Channel.js#L156

How much info is available here?

Reply:
For `CHANNEL_UPDATE` it's only `name` but for the other two events, it's:

![uR1MafnvnC](https://user-images.githubusercontent.com/51517103/121218461-61703c00-c8a0-11eb-9a27-51d7e976622f.png)

Agreed, for the delete event caching the channel on create isn't a good idea. Maybe, we can add an extra boolean param to `Channel#create` to control caching. It will be true by default so it won't break anything that depends on the `Channel#create` method.

Reply:
Thanks, for the thumbs up kyra (I saw it after the commit 😸) but now I feel like all the above effort is not worth for just getting an object for a deleted channel with partial properties. So, I removed the channel creation part. Now, it is just an object with whatever data the api sends in `changes`. For cached channel it will still be a complete channel object.